src/node-file-trace.ts

```diff
@@ -12,6 +12,30 @@ const fsReadFile = fs.promises.readFile;
 const fsReadlink = fs.promises.readlink;
 const fsStat = fs.promises.stat;
 
+type ParseSpecifierResult = {
+  path: string;
+  queryString: string | null
+}
+
+// Splits an ESM specifier into path and querystring (including the leading `?`). (If the specifier is CJS,
+// it is passed through untouched.)
+export function parseSpecifier(specifier: string, cjsResolve: boolean = true): ParseSpecifierResult {
```

Comment:
```suggestion
export function parseSpecifier(specifier: string, cjsResolve: boolean): ParseSpecifierResult {
```

Lets not use a default param or else the caller might forget

Reply:
Actually, we need the default to be there, because the first time this is called (at the top of `analyzeAndEmitDependency` when it's called by `emitDependency`), we don't have a `cjsResolve` value. Having it default to `true` works in that case, though, because that only happens at the root of the tree, when `nodeFileTrace` calls `emitDependency` on an actual file (not an import specifier), which we know won't have a querystring.